pinot-common/src/main/java/org/apache/pinot/common/restlet/resources/StartReplaceSegmentsRequest.java

```diff
@@ -37,11 +37,11 @@ public class StartReplaceSegmentsRequest {
   private final List<String> _segmentsTo;
 
   public StartReplaceSegmentsRequest(@JsonProperty("segmentsFrom") @Nullable List<String> segmentsFrom,
-      @JsonProperty("segmentsTo") List<String> segmentsTo) {
+      @JsonProperty("segmentsTo") @Nullable List<String> segmentsTo) {
     _segmentsFrom = (segmentsFrom == null) ? Collections.emptyList() : segmentsFrom;
-    _segmentsTo = segmentsTo;
-    Preconditions
-        .checkArgument(segmentsTo != null && !segmentsTo.isEmpty(), "'segmentsTo' should not be null or empty");
+    _segmentsTo = (segmentsTo == null) ? Collections.emptyList() : segmentsTo;
+    Preconditions.checkArgument(!_segmentsFrom.isEmpty() || !_segmentsTo.isEmpty(),
+        "Both segmentsFrom and segmentsTo cannot be empty");
```

Comment:
(nit) This sentence is a little bit confusing :-P
```suggestion
        "'segmentsFrom' and 'segmentsTo' cannot both be empty");
```

Reply:
lol this is a side effect of Ai. Git `copilot` recommended the above sentence :P